skcosmo/preprocessing/_flexible_scaler.py

```diff
@@ -0,0 +1,600 @@
+    ):
+        """Initialize StandardFlexibleScaler."""
+        self.with_mean = with_mean
+        self.with_std = with_std
+        self.column_wise = column_wise
+        self.rtol = rtol
+        self.atol = atol
+        self.n_samples_seen_ = 0
+
+    def fit(self, X, y=None, sample_weight=None):
+        """Compute mean and scaling to be applied for subsequent normalization.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features)
+            The data used to compute the mean and standard deviation
+            used for later scaling along the features axis.
+
+        y: None
+            Ignored.
+
+        sample_weight: ndarray of shape (n_samples,)
+            Weights for each sample. Sample weighting can be used to center
+            (and scale) data using a weighted mean. Weights are internally
+            normalized before preprocessing.
+
+        Returns
+        -------
+        self : object
+            Fitted scaler.
+        """
+
+        self.n_samples_seen_, self.n_features_ = X.shape
+
+        if sample_weight is not None:
+            sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
+            sample_weight = sample_weight / np.sum(sample_weight)
+
+        if self.with_mean:
+            self.mean_ = np.average(X, weights=sample_weight, axis=0)
+        else:
+            self.mean_ = np.zeros(self.n_features_)
+
+        self.scale_ = 1.0
+        if self.with_std:
+            X_mean = np.average(X, weights=sample_weight, axis=0)
+            var = np.average((X - X_mean) ** 2, weights=sample_weight, axis=0)
+
+            if self.column_wise:
+                if np.any(var < self.atol + abs(X_mean) * self.rtol):
+                    raise ValueError("Cannot normalize a feature with zero variance")
+                self.scale_ = np.sqrt(var)
+            else:
+                var_sum = var.sum()
+                if var_sum < abs(np.mean(X_mean)) * self.rtol + self.atol:
+                    raise ValueError("Cannot normalize a matrix with zero variance")
+                self.scale_ = np.sqrt(var_sum)
+
+        return self
+
+    def transform(self, X, y=None):
+        """Normalize a vector based on previously computed mean and scaling.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features)
+            The data used to scale along the features axis.
+
+        y: None
+            Ignored.
+
+        Returns
+        -------
+        X_tr : ndarray of shape (n_samples, n_features)
+            Transformed array.
+        """
+
+        check_is_fitted(self, attributes=["n_samples_seen_", "n_features_"])
+
+        if self.n_features_ != X.shape[1]:
+            raise ValueError("X shape does not match training shape")
+        return (X - self.mean_) / self.scale_
+
+    def fit_transform(self, X, y=None, sample_weight=None, **fit_params):
```

Comment:
Is it necessary to use `**fit_params` opposed to `**kwargs`?

Reply:
There is essentially no difference. But `**fit_params` is used everywhere in `sklearn` for `fit_transform`, so I don't see any reason to deviate from these notations.